crates/bevy_pbr/src/render/pbr_lighting.wgsl

```diff
@@ -284,3 +285,151 @@ fn directional_light(light_id: u32, roughness: f32, NdotV: f32, normal: vec3<f32
 
     return (specular_light + diffuse) * (*light).color.rgb * NoL;
 }
+
+fn transmissive_light(world_position: vec4<f32>, frag_coord: vec3<f32>, N: vec3<f32>, V: vec3<f32>, ior: f32, thickness: f32, perceptual_roughness: f32, transmissive_color: vec3<f32>, transmitted_environment_light_specular: vec3<f32>) -> vec3<f32> {
+    // Calculate distance, used to scale roughness transmission blur
```

Comment:
Rather then distance from the camera, do you mean to use screen-space depth? Right now, an object far away from the camera depth-wise, and an object far to the left of the camera have the same distance. That doesn't make sense for perspective projections.

Reply:
Hmmm yeah, nice catch. This type of distance would only make sense for something like a 180º FoV fisheye lens.